### ISSUE TITLE
[BUG] float31sign truncates high digits when input exceeds -99..+99

### ISSUE THREAD

**Issue by vlsi** [MarlinFirmware/Marlin#26262]
### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

See https://github.com/MarlinFirmware/Marlin/blob/11f98adcce51f06a85ef0dd64e6e6b16139a34bd/Marlin/src/lcd/menu/menu_advanced.cpp#L631-L632

ColorUI editor for Z-probe X offset and Z-probe Y offset ignore high digits exceeding +-99.

Here's how it goes:
1) Editor uses -(X_BED_SIZE), X_BED_SIZE bounds, and it uses float31sign formatting
2) In my case X_BED_SIZE is 325, and float31sign performs `12.3`-like formatting (see https://github.com/MarlinFirmware/Marlin/blob/11f98adcce51f06a85ef0dd64e6e6b16139a34bd/Marlin/src/lcd/menu/menu_item.h#L158 )
3) Unfortunately, `ftostr31sign` just uses `mod 100`, so if I slide the editor to the max (~`325`), then UI shows just `25` which is misleading.

----

I suggest adding checks to `ftostr31sign`, `ftostr41sign` and similar functions so a call to `ftostr31sign` would forward the call to `ftostr41sign` if the value exceeds -99..+99.

An alternative option would be to format the value right to left like in `i16tostr3left`. In that case, we could have a single function for both `31` and `41` formatters.

An alternative opinion is to enforce `3 digit` size and use something like `##.#` when value exceeds `-99..+99`, however, I think displaying one more char is better than displaying none of the chars at all.

### Expected behavior

I expect the editor and formatter to show when the value exceeds the expected bounds.
For instance, even if "x offset" is supposed to be smallish like `12.3`, I expect Marlin to format it as `319.2` rather than `19.2` if the actual value is `319.2`.

### Actual behavior

Marlin ignores high digits, and it is impossible to tell that from the UI

### Steps to Reproduce

1. Set `X_BED_SIZE` to 325 (or whatever exceeding 99)
1. Start editing z-probe x offset in ColorUI
1. Select "maximal value"

ER: `x offset` should be displayed as three-digit float
AR: only the two last digits are shown

### Version of Marlin Firmware

2.1.2.1

### Bed Leveling

None

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

See configs in https://github.com/vlsi/reborn2-marlin/tree/v2.1.2.1-2023-09-08_1932/Marlin

**Comment by ellensp** [comment]
The idea of float31sign is to constrain the result to the digits expected, so the lcd does not get all messed up
It is not meant to auto range or scale

These menus where the value is truncated should be upgraded to float41sign and recheck the display is ok.  (on a 128x64 display, ColorUI is using same code.) 

An assert should probably be generated when the max and min exceed the requested display type

**Comment by vlsi** [comment]
Currently, float31sign corrupts output, and it is hard to tell when the value does not fit

**Comment by ellensp** [comment]
no, The DEFINE_MENU_EDIT_ITEM_TYPE code does what its is meant to do . Ie force the result into a particular string size

It uses ftostr31sign

```CPP
// Convert float to fixed-length string with +12.3 / -12.3 format
const char* ftostr31sign(const_float_t f) {
  int i = INTFLOAT(f, 1);
  conv[3] = MINUSOR(i, '+');
  conv[4] = DIGIMOD(i, 100);
  conv[5] = DIGIMOD(i, 10);
  conv[6] = '.';
  conv[7] = DIGIMOD(i, 1);
  return &conv[3];
}

```
ie the result will be in 5 characters, always
consisting of:
 
a single character representing the sign.  '-' or '+'
2 digits   00 - 99
a period  '.'
1 digit  0-9


The EDIT_ITEM code needs to use float41sign to accept 3 digits, 000-999
